things/service.go

```diff
@@ -247,13 +242,46 @@ func (ts *thingsService) UpdateThing(ctx context.Context, token string, thing Th
 	return ts.things.Update(ctx, thing)
 }
 
+func (ts *thingsService) ShareThing(ctx context.Context, token, thingID string, policies, userIDs []string) error {
+	res, err := ts.auth.Identify(ctx, &mainflux.Token{Value: token})
+	if err != nil {
+		return errors.Wrap(ErrUnauthorizedAccess, err)
+	}
+
+	if err := ts.authorize(ctx, res.GetId(), thingID, writeRelationKey); err != nil {
+		return err
+	}
+
+	return ts.claimOwnership(ctx, thingID, policies, userIDs)
+}
+
+func (ts *thingsService) claimOwnership(ctx context.Context, thingID string, policies, userIDs []string) error {
+	var errs error
+	for _, userID := range userIDs {
+		for _, policy := range policies {
+			apr, err := ts.auth.AddPolicy(ctx, &mainflux.AddPolicyReq{Obj: thingID, Act: policy, Sub: userID})
+			if err != nil {
+				errs = accumulateError(errs, errors.Wrap(ErrAuthorization, err), thingID, userID)
+			}
+			if !apr.GetAuthorized() {
+				errs = accumulateError(errs, ErrAuthorization, thingID, userID)
+			}
+		}
+	}
+	return errs
+}
+
+func accumulateError(errs, newErr error, thingID, userID string) error {
+	return errors.Wrap(newErr, errors.Wrap(fmt.Errorf("cannot claim ownership on thing: %s by user: %s", thingID, userID), errs))
+}
```

Comment:
No need for a separate function and there is no need to wrap each error to `ErrAuthorization`. Something like:
```go
func (ts *thingsService) claimOwnership(ctx context.Context, thingID string, policies, userIDs []string) error {
	var errs error
	for _, userID := range userIDs {
		for _, policy := range policies {
			apr, err := ts.auth.AddPolicy(ctx, &mainflux.AddPolicyReq{Obj: thingID, Act: policy, Sub: userID})
			if err != nil {
                                errs = errors.Wrap(fmt.Errorf("Cannot claim ownership on thing '%s' by user '%s': %s", thingID, userID, err), errs))
			}
			if !apr.GetAuthorized() {
				errs = errors.Wrap(fmt.Errorf("Cannot claim ownership on thing '%s' by user '%s': unauthorized", thingID, userID), errs))
			}
		}
	}
        if errs != nil {
                return errors.Wrap(ErrAuthorization, errs)
        }
	return nil
}
```
will do.